packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -200,4 +245,54 @@ export class HttpApi extends Resource implements IHttpApi {
       integration: options.integration,
     }));
   }
+
+  /**
+   * Add a custom domain for this API and create the API mapping to a stage.
+   */
+  public addDomainName(options: AddDomainNameOptions): DomainName {
+    const dn = new DomainName(this, `DomainName${options.domainName}`, {
+      domainName: options.domainName,
+      certificate: options.certificate,
+    });
+
+    // const stage = options.stage ?? this.defaultStage!;
+
+    // const mapping = new HttpApiMapping(this, `${dn.domainName}defaultMapping`, {
+    //   api: this,
+    //   domainName: dn,
+    //   stage,
+    // });
+    // mapping.node.addDependency(stage);
+
+    this._domainName = dn;
+
+    return dn;
+  }
+
+  /**
+   * Create the `HttpApiMapping` resource with the custom domain and stage for the HTTP API
+   * @param options Options for domain name mapping
+   */
+  public addDomainMapping(options: DomainMappingOptions) {
```

Comment:
Drop this API as well. 

Domain mappings are added to Stages and not to `HttpApi`. Even with a Stage, they are configured once and cannot be added multiple times. This API is not required.